service/controller/clusterapi/v29/resource/ipam/create_test.go

```diff
@@ -0,0 +1,109 @@
+package ipam
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/micrologger/microloggertest"
+	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+func Test_SubnetAllocator(t *testing.T) {
```

Comment:
The resource implementation as well as the unit tests are ported from the `network` package. I am going to drop it as the resource here replaces it now properly with additional capabilities of working with different types. 

Note that the tests are different in error handling as the matching for the allocated subnets is handled differently. It didn't appear to make sense to test this. When you think differently, this test should be extensible to do it rather easily.

Reply:
Not deep into this topic but that makes sense to me.